chore(parental-leave): Update readme (#5458)
island-is/island.is: libs/application/templates/parental-leave/README.md

# Parental leave application

## Description

> This application template allows applicants to apply for parental leave. For further reading about application templates, have a look at `libs/application/templates/reference-template/README.md`.

Reviewer: Should be a relative link?

https://docs.devland.is/repository/gitbook-contribute#links

- It contains all the steps for parents and employers required to collect and confirm all information
- It handles communications with external APIs to filter out valid applicants as well as sending complete applications to Vinnumálastofnun

Below you'll find a flow chart for the application

[Flow chart]

### Glossary

| Word             | Description                   |
| ---------------- | ----------------------------- |
| Primary parent   | The mother carrying the child |
| Secondary parent | The other parent              |
| VMST             | Vinnumálastofnun              |

### States

> This section assumes that you are familiar with [states](https://docs.devland.is/libs/application/core#states).

Reviewer: Should be relative `../path/to/README.md#states`

#### Other parent approval

When the primary parent requests transferal of rights, then the other secondary parent will need to approve of the transfer. The other parent will receive an email with a link to approve of this request.

#### Employer waiting to assign

If the applicant is employed by an employer (not self employed), then they are asked to provide an email address which a confirmation email will be sent to. When the employer receives the email they can click a link which automatically assigns them to the application and advances it to the next state

#### Employer approval

Here the employer will have a chance to review the periods selected by the applicant as well as pension fund and union.

The employer can then approve or request changes

#### Vinnumalastofnun approval

Applications that have been sent to VMST

### Parental leave template API module

For async actions that require server side logic we have the template api modules. The module for this application contains email templates as well as external actions like `sendApplication`.

> Have a look at `libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.module.ts` for further information on these actions, and `libs/application/template-api-modules/README.md` to get up to speed on template api modules.

Reviewer: Relative links?

### API and X-Road

> All user interaction goes through our GraphQL API (`api`) which then handles communicating with other APIs.

Reviewer: ```suggestion
All user interactions go through our GraphQL API (`api`) which is integrated with with other APIs.
```

> To communicate with VMST a request has to go through [X-Road](https://docs.devland.is/technical-overview/x-road). Both in development and on production environments.

Reviewer: Relative link?

To connect to VMST test API you'll want to start the local proxy, following these steps:

- Open AWS: https://island-is.awsapps.com/start#/
  - island-is-development01 > AWSPowerUserAccess > Command line access
- Choose an option you prefer in the modal

> - Connect to the proxy `./scripts/run-xroad-proxy.sh`

Reviewer: This is kind of documented [here](https://docs.devland.is/#running-proxy-against-development-service) and [here](https://docs.devland.is/repository/aws-secrets#using-aws-session).

> See `libs/api/domains/directorate-of-labour/src/lib/directorate-of-labour.module.ts` for examples of VMST communication.

Reviewer: Relative link?

### Contacts

VMST API specific questions:
[Guðmundur K. G. Kolka](mailto:gudmundur.kolka@advania.is), API and NAV developer

VMST Devops:

> [Unnar Sigurðsson](mailto:unnar.sigurdsson@vmst.is), VMST API dev ops and x-road integration

Reviewer: Should probably keep this kind of stuff out of the repo. Maybe link to [Notion](https://www.notion.so/Vinnum-lastofnun-c7bc135b70ad431e846d413846a60967) instead, and specify "Technical Contacts" there.

We also have two channels on the islandis slack #faedingarorlof-profanir and #faedingarorlof

### Localisation

All localisation can be found on Contentful.

- [Parental leave application translations](https://app.contentful.com/spaces/8k0h54kbe6bj/entries/pl.application)
- [Application system translations](https://app.contentful.com/spaces/8k0h54kbe6bj/entries/application.system)

> ⚠️ When creating new text strings in the messages.ts file for the application, be sure to update Contentful, this is done using the command:

Reviewer: Maybe wrap this section like this instead:

```
{% hint style="warning" %}
Content
{% endhint %}
```

```
CONTENTFUL_MANAGEMENT_ACCESS_TOKEN={secret*} yarn nx run application-templates-parental-leave:extract-strings
```

_secret is found within AWS parameter store_

> See [link](https://docs.devland.is/apps/web/subpages#contentful) for further documentation for localisation in island.is repo.

Reviewer: Bad link?

Maybe link to [this](https://docs.devland.is/libs/localization#message-extraction) and leave out the CONTENTFUL_MANAGEMENT_ACCESS_TOKEN stuff.

### Emails

As previously mentioned, the application sends out emails to applicants and assignees. The email templates are stored in the parental leave api template module.

When developing locally you’ll see preview links for would-be emails where you can see how they would be rendered and click links.

## Setup

See [application-system](https://docs.devland.is/apps/application-system) setup on how to get started.

You'll want to start the `application-system-api` and `api` with these env variables to form the correct X-Road requests:

- `XROAD_BASE_PATH_WITH_ENV=http://localhost:8081/r1/IS-DEV`
- `XROAD_VMST_MEMBER_CODE=10003`
- `XROAD_VMST_API_PATH=/VMST-ParentalLeave-Protected/ParentalLeaveApplication-v1`
- `XROAD_CLIENT_ID=IS-DEV/GOV/10000/island-is-client`

Then you'll also want these env variables:

- `VMST_API_KEY`: to communicate with the VMST API, found in AWS parameter store

> - `CONTENTFUL_ACCESS_TOKEN`: to fetch translations, found in Contentful

Reviewer: Would be nice to list commands to fetch these with [get-secrets](https://docs.devland.is/repository/aws-secrets#usage-to-fetch-secrets)

Then also start `application-system-form`.

Once you have everything running you can navigate to `http://localhost:4200/umsoknir/faedingarorlof` and start developing.

### Local database

> By setting up the application-system you'll have created a local postgres database on a docker image, if you haven't already it is recommended to a tool to interact with your database. For example [pgAdmin](https://www.pgadmin.org/download/).

Reviewer: ```suggestion
By setting up the application-system you'll have created a local postgres database on a docker image, if you haven't already you should setup a tool to interact with your database. For example [pgAdmin](https://www.pgadmin.org/download/).
```
